### PR TITLE
Upgrade to v8

### PR DESCRIPTION
**Pull request by freekmurze**

### CODE REVIEW

File: src/UrlGenerator/BaseUrlGenerator.php [hunk 1]
```diff
@@ -1,84 +1,70 @@
...
         if (is_null($this->conversion)) {
             return $this->pathGenerator->getPath($this->media).($this->media->file_name);
         }
 
         return $this->pathGenerator->getPathForConversions($this->media)
-                .$this->conversion->getConversionFile($this->media->file_name);
+                .$this->conversion->getConversionFile($this->media);
     }
 
-    public function rawUrlEncodeFilename(string $path = ''): string
```
Comment (brendt) [inline src/UrlGenerator/BaseUrlGenerator.php]: I believe this removal should be listed in UPGRADING

File: tests/Feature/Conversion/ConversionFileNamerTest.php [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+use Spatie\Medialibrary\Tests\Support\testfiles\TestConversionFileNamer;
+use Spatie\Medialibrary\Tests\TestCase;
+
+class ConversionFileNamerTest extends TestCase
+{
+    /** @test */
+    public function it_can_use_a_custom_conversion_file_namer()
+    {
+        config()->set('medialibrary.conversion_file_namer', TestConversionFileNamer::class);
```
Comment (brendt) [inline tests/Feature/Conversion/ConversionFileNamerTest.php]: My preference would be to add this `TestConversionFileNamer` within this test file, if it's not used anywhere else. No need to making separate files for testing edge cases.

Another approach would be to use an anonymous class.

Comment (brendt) [reply tests/Feature/Conversion/ConversionFileNamerTest.php]: A little more background information about why I feel this way: you could end up with lots of classes all doing something very specific for one or two tests. I find this adds more confusion over time than anything else. The states package is an example of where this grew too complex: https://github.com/spatie/laravel-model-states/tree/master/tests/Dummy

File: CHANGELOG.md [hunk 3]
```diff
@@ -2,6 +2,21 @@
...
+- added `uuid` on `media` table
+- an empty string is now a proper collection name. `getMedia('')` will not return media from the default collection anymore (#1697).
+- add the ability to store conversions on a separate disk
+- simplify URL generation. You can now just use the `root` and `url` properties on a configured disk
+- spatie/pdf-to-image is now a suggestion dependency, removing the need for always having to install ext-imagick
+- added `shouldMatchBothExtensionsAndMimeTypes` to `Spatie\Medialibrary\ImageGenerators\BaseGenerator`
+- added progress bar on the clean command (#1623)
+- the `UrlGenerator` interface now contains all required methods (#1656)
+- use PHP 7.4 features where possible
+- the namespace has been renamed from `Spatie\MediaLibrary` to `Spatie\Medialibrary`. All code where `MediaLibrary` was used is renamed to `Medialibrary`
```
[the remark this answers is not in the document]
Comment (freekmurze) [reply CHANGELOG.md]: It reflects the package name better.

File: database/migrations/create_media_table.php.stub [hunk 4]
```diff
@@ -6,19 +6,18 @@ use Illuminate\Database\Migrations\Migration;
...
 {
-    /**
-     * Run the migrations.
-     */
     public function up()
     {
         Schema::create('media', function (Blueprint $table) {
             $table->bigIncrements('id');
             $table->morphs('model');
+            $table->string('uuid')->nullable();
```
[the remark this answers is not in the document]
Comment (freekmurze) [reply database/migrations/create_media_table.php.stub]: Not really, changed it

File: docs/converting-images/naming-conversion-files.md [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+weight: 4
+---
+
+By default, all conversion files will be named in this format:
+
+```
+{original-file-name-without-extension}-{name-of-the-conversion}.{extension}
+```
+
+Should you want to name your conversion file using another format, than you can specify your the class name of your own `ConversionFileNamer` in the `conversion_file_namer` key of the `medialibrary.php` config file.
```
[the remark this answers is not in the document]
Comment (freekmurze) [reply docs/converting-images/naming-conversion-files.md]: I don't see the typo 🙂 So yeah, please fix it 👍

File: src/UrlGenerator/BaseUrlGenerator.php (hunk 1 above)
Comment (freekmurze) [reply src/UrlGenerator/BaseUrlGenerator.php]: Done

File: tests/Feature/Conversion/ConversionFileNamerTest.php (hunk 2 above)
Comment (freekmurze) [reply tests/Feature/Conversion/ConversionFileNamerTest.php]: I'm going to leave it like it is for now. I plan on doing a big refactor on the tests after release, and will address this at that time. Thanks for the suggestion.

File: docs/converting-images/naming-conversion-files.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (brendt) [reply docs/converting-images/naming-conversion-files.md]: > your the class name 

It's fixed ;)

[the remark this answers is not in the document]
Comment (freekmurze) [reply docs/converting-images/naming-conversion-files.md]: Thanks!

File: src/Conversion/Conversion.php [hunk 6]
```diff
@@ -265,13 +213,20 @@ public function getResultExtension(string $originalFileExtension = ''): string
         return $originalFileExtension;
     }
 
-    public function getConversionFile(string $file): string
+    public function getConversionFile(Media $media): string
```
[the remark this answers is not in the document]
Comment (freekmurze) [reply src/Conversion/Conversion.php]: Done

File: src/Models/Media.php [hunk 7]
```diff
@@ -287,14 +275,20 @@ public function img($conversion = '', array $extraAttributes = []): string
...
-            })->implode(' ');
+            ->map(fn($value, $name) => $name.'="'.$value.'"')->implode(' ');
 
         if (strlen($attributeString)) {
             $attributeString = ' '.$attributeString;
         }
 
+        $loadingAttributeValue = config('medialibrary.default_loading_attribute_value');
+        if ($conversion !== '') {
+            /** @var Conversion $conversionObject */
```
[the remark this answers is not in the document]
Comment (freekmurze) [reply src/Models/Media.php]: 👍

File: src/Conversion/DefaultConversionFileNamer.php [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+    public function getName(Conversion $conversion, Media $media): string
+    {
+        $fileName = pathinfo($media->file_name, PATHINFO_FILENAME);
+        
+        $extension = $this->getExtension($conversion, $media);
+
+        return "{$fileName}-{$conversion->getName()}.{$extension}";
+    }
+    
+    public function getExtension(Conversion $conversion, Media $media): string
```
[the remark this answers is not in the document]
Comment (freekmurze) [reply src/Conversion/DefaultConversionFileNamer.php]: Nice catch!

File: src/MedialibraryServiceProvider.php [hunk 9]
```diff
@@ -1,16 +1,16 @@
...
-
-class MediaLibraryServiceProvider extends ServiceProvider
+use Spatie\Medialibrary\Commands\CleanCommand;
+use Spatie\Medialibrary\Commands\ClearCommand;
+use Spatie\Medialibrary\Commands\RegenerateCommand;
+use Spatie\Medialibrary\Filesystem\Filesystem;
+use Spatie\Medialibrary\ResponsiveImages\TinyPlaceholderGenerator\TinyPlaceholderGenerator;
+use Spatie\Medialibrary\ResponsiveImages\WidthCalculator\WidthCalculator;
+
+class MedialibraryServiceProvider extends ServiceProvider
```
[the remark this answers is not in the document]
Comment (freekmurze) [reply src/MedialibraryServiceProvider.php]: Done

File: src/Models/Media.php [hunk 10]
```diff
@@ -10,28 +10,30 @@
...
-        CustomMediaProperties;
+        CustomMediaProperties,
+        HasUuid;
 
     const TYPE_OTHER = 'other';
 
     protected $guarded = [];
 
-    protected $casts = [
+    protected  $casts = [
```
[the remark this answers is not in the document]
Comment (freekmurze) [reply src/Models/Media.php]: Wiped from existence.

File: src/Jobs/GenerateResponsiveImagesJob.php [hunk 11]
```diff
@@ -1,20 +1,19 @@
...
 use Illuminate\Contracts\Queue\ShouldQueue;
 use Illuminate\Queue\InteractsWithQueue;
 use Illuminate\Queue\SerializesModels;
-use Spatie\MediaLibrary\Models\Media;
-use Spatie\MediaLibrary\ResponsiveImages\ResponsiveImageGenerator;
+use Spatie\Medialibrary\Models\Media;
+use Spatie\Medialibrary\ResponsiveImages\ResponsiveImageGenerator;
 
-class GenerateResponsiveImages implements ShouldQueue
+class GenerateResponsiveImagesJob implements ShouldQueue
```
[the remark this answers is not in the document]
Comment (freekmurze) [reply src/Jobs/GenerateResponsiveImagesJob.php]: I trust that nobody will have used this directly. Not adding it to the upgrade guide at this moment.

File: src/ImageGenerators/BaseGenerator.php [hunk 12]
```diff
@@ -13,14 +13,18 @@ public function canConvert(Media $media): bool
...
 
-        if ($this->supportedExtensions()->contains(strtolower($media->extension))) {
-            return true;
-        }
+        $validExtension = $this->canHandleExtension(strtolower($media->extension));
+        $validMimeType = $this->canHandleMime(strtolower($media->mime_type));
 
-        if ($this->supportedMimetypes()->contains(strtolower($media->mime_type))) {
-            return true;
+        if ($this->shouldMatchBothExtensionsAndMimeTypes()) {
```
[the remark this answers is not in the document]
Comment (freekmurze) [reply src/ImageGenerators/BaseGenerator.php]: This function determines how the check should be performed, it's not the result of a check. So, I think the current name is better.

File: src/FileManipulator.php [hunk 13]
```diff
@@ -72,7 +70,7 @@ public function performConversions(ConversionCollection $conversions, Media $med
 
         $temporaryDirectory = TemporaryDirectory::create();
 
-        $copiedOriginalFile = app(Filesystem::class)->copyFromMediaLibrary(
+        $copiedOriginalFile = app(Filesystem::class)->copyFromMedialibrary(
```
[the remark this answers is not in the document]
Comment (freekmurze) [reply src/FileManipulator.php]: There's already

```
- replace all code occurrences of `mediaLibrary` by `medialibrary`
```

that covers this.

File: src/FileAdder/FileAdder.php [hunk 14]
```diff
@@ -290,11 +277,14 @@ public function toMediaCollection(string $collectionName = 'default', string $di
         $media->file_name = $this->fileName;
 
         $media->disk = $this->determineDiskName($diskName, $collectionName);
-
         if (is_null(config("filesystems.disks.{$media->disk}"))) {
             throw DiskDoesNotExist::create($media->disk);
         }
 
+        $media->conversions_disk = $this->determineConversionsDiskName($media->disk, $collectionName);
+        if (is_null(config("filesystems.disks.{$media->conversions_disk}"))) {
```
[the remark this answers is not in the document]
Comment (freekmurze) [reply src/FileAdder/FileAdder.php]: I've streamlined it a bit by moving the duplicated check to it's own function. I haven't put the check in `determineConversionsDiskName`, because then I'd lose the ability to early return.

If you feel this can be improved, feel free to commit it to this branch.

File: src/Conversion/DefaultConversionFileNamer.php [hunk 15]
```diff
@@ -0,0 +1,24 @@
...
+
+namespace Spatie\Medialibrary\Conversion;
+
+use Spatie\Medialibrary\Models\Media;
+
+class DefaultConversionFileNamer implements ConversionFileNamer
+{
+    public function getName(Conversion $conversion, Media $media): string
+    {
+        $fileName = pathinfo($media->file_name, PATHINFO_FILENAME);
```
[the remark this answers is not in the document]
Comment (freekmurze) [reply src/Conversion/DefaultConversionFileNamer.php]: I've changed this so a user should only implement `getFileName` and does not need to care about an extension.

I'd love to have a `File` value object in the code base, but it seems a lot of effort for small gains. If you think you can implement this quickly, do send a PR!

File: src/FileAdder/FileAdder.php (hunk 14 above)
[the remark this answers is not in the document]
Comment (brendt) [reply src/FileAdder/FileAdder.php]: > because then I'd lose the ability to early return.

My thinking was that you'd throw from within `determineConversionsDiskName`, which would in essence be an early return without the clutter. But no problem to keep it like it is now!

File: src/Conversions/Jobs/PerformConversionsJob.php [hunk 16]
```diff
@@ -0,0 +1,41 @@
...
+{
+    use InteractsWithQueue, SerializesModels, Queueable;
+
+    protected ConversionCollection $conversions;
+
+    protected Media $media;
+
+    protected bool $onlyMissing;
+
+    public function __construct(ConversionCollection $conversions, Media $media, $onlyMissing = false)
```
Comment (AlexVanderbist) [inline src/Conversions/Jobs/PerformConversionsJob.php]: Boolean typehint on `$onlyMissing`

File: src/Conversions/Jobs/PerformConversionsJob.php [hunk 17]
```diff
@@ -0,0 +1,41 @@
...
+
+        $this->media = $media;
+
+        $this->onlyMissing = $onlyMissing;
+    }
+
+    public function handle(): bool
+    {
+        /** @var \Spatie\MediaLibrary\Conversions\FileManipulator $fileManipulator */
+        $fileManipulator = app(FileManipulator::class);
```
Comment (AlexVanderbist) [inline src/Conversions/Jobs/PerformConversionsJob.php]: `$fileManipulator` can be resolved from container in `handle` method

File: src/MediaCollections/Commands/CleanCommand.php [hunk 18]
```diff
@@ -1,72 +1,55 @@
...
+use Spatie\MediaLibrary\MediaCollections\Models\Media;
+use Spatie\MediaLibrary\Support\PathGenerator\DefaultPathGenerator;
 use Spatie\MediaLibrary\ResponsiveImages\RegisteredResponsiveImages;
 
 class CleanCommand extends Command
 {
     use ConfirmableTrait;
 
-    protected $signature = 'medialibrary:clean {modelType?} {collectionName?} {disk?}
+    protected $signature = 'media-library:clean {modelType?} {collectionName?} {disk?}
```
Comment (AlexVanderbist) [inline src/MediaCollections/Commands/CleanCommand.php]: Document change in UPGRADING.md

File: src/MediaCollections/Commands/ClearCommand.php [hunk 19]
```diff
@@ -1,37 +1,42 @@
...
-use Spatie\MediaLibrary\MediaRepository;
+use Spatie\MediaLibrary\MediaCollections\MediaRepository;
+use Spatie\MediaLibrary\MediaCollections\Models\Media;
 
 class ClearCommand extends Command
 {
     use ConfirmableTrait;
 
-    protected $signature = 'medialibrary:clear {modelType?} {collectionName?}
+    protected $signature = 'media-library:clear {modelType?} {collectionName?}
```
Comment (AlexVanderbist) [inline src/MediaCollections/Commands/ClearCommand.php]: Change missing in upgrading.md

File: src/MediaCollections/Exceptions/DiskDoesNotExist.php [hunk 20]
```diff
@@ -0,0 +1,11 @@
+<?php
+
+namespace Spatie\MediaLibrary\MediaCollections\Exceptions;
+
+class DiskDoesNotExist extends FileCannotBeAdded
+{
+    public static function create($diskName): self
```
Comment (AlexVanderbist) [inline src/MediaCollections/Exceptions/DiskDoesNotExist.php]: string typehint on `$diskName`

File: src/MediaCollections/Exceptions/InvalidPathGenerator.php [hunk 21]
```diff
@@ -0,0 +1,21 @@
...
+    public static function doesntExist(string $class): self
+    {
+        return new static("Path generator class {$class} doesn't exist");
+    }
+
+    public static function doesNotImplementPathGenerator(string $class): self
+    {
+        $pathGeneratorClass = PathGenerator::class;
+
+        return new static("Path generator class {$class} must implement `$pathGeneratorClass}`");
```
Comment (AlexVanderbist) [inline src/MediaCollections/Exceptions/InvalidPathGenerator.php]: Backticks around classname?

File: src/MediaCollections/Exceptions/InvalidPathGenerator.php [hunk 22]
```diff
@@ -0,0 +1,21 @@
...
+namespace Spatie\MediaLibrary\MediaCollections\Exceptions;
+
+use Exception;
+use Spatie\MediaLibrary\Support\PathGenerator\PathGenerator;
+
+class InvalidPathGenerator extends Exception
+{
+    public static function doesntExist(string $class): self
+    {
+        return new static("Path generator class {$class} doesn't exist");
```
Comment (AlexVanderbist) [inline src/MediaCollections/Exceptions/InvalidPathGenerator.php]: Backticks around classname?

File: src/MediaCollections/Exceptions/MediaCannotBeDeleted.php [hunk 23]
```diff
@@ -1,13 +1,13 @@
...
-namespace Spatie\MediaLibrary\Exceptions;
+namespace Spatie\MediaLibrary\MediaCollections\Exceptions;
 
 use Exception;
 use Illuminate\Database\Eloquent\Model;
 
 class MediaCannotBeDeleted extends Exception
 {
-    public static function doesNotBelongToModel($mediaId, Model $model)
+    public static function doesNotBelongToModel($mediaId, Model $model): self
```
Comment (AlexVanderbist) [inline src/MediaCollections/Exceptions/MediaCannotBeDeleted.php]: int typehint on `$mediaId`?

File: src/MediaCollections/Exceptions/MimeTypeNotAllowed.php [hunk 24]
```diff
@@ -1,12 +1,10 @@
...
 
-namespace Spatie\MediaLibrary\Exceptions\FileCannotBeAdded;
-
-use Spatie\MediaLibrary\Exceptions\FileCannotBeAdded;
+namespace Spatie\MediaLibrary\MediaCollections\Exceptions;
 
 class MimeTypeNotAllowed extends FileCannotBeAdded
 {
-    public static function create($file, array $allowedMimeTypes)
+    public static function create($file, array $allowedMimeTypes): self
```
Comment (AlexVanderbist) [inline src/MediaCollections/Exceptions/MimeTypeNotAllowed.php]: string typehint on `$file`?

File: src/MediaCollections/FileAdder.php [hunk 25]
```diff
@@ -176,6 +156,13 @@ public function withCustomProperties(array $customProperties): self
         return $this;
     }
 
+    public function storingConversionsOnDisk(string $diskName)
```
Comment (AlexVanderbist) [inline src/MediaCollections/FileAdder.php]: `self` returntype

File: src/MediaCollections/Filesystem.php [hunk 26]
```diff
@@ -133,66 +150,65 @@ public function copyFromMediaLibrary(Media $media, string $targetFile): string
...
         $responsiveImagesDirectory = $this->getMediaDirectory($media, 'responsiveImages');
 
-        collect([$mediaDirectory, $conversionsDirectory, $responsiveImagesDirectory])
+        $this->filesystem->disk($media->disk)->deleteDirectory($mediaDirectory);
 
+        collect([$mediaDirectory, $conversionsDirectory, $responsiveImagesDirectory])
             ->each(function ($directory) use ($media) {
-                $this->filesystem->disk($media->disk)->deleteDirectory($directory);
+                $this->filesystem->disk($media->conversions_disk)->deleteDirectory($directory);
             });
```
Comment (AlexVanderbist) [inline src/MediaCollections/Filesystem.php]: `$mediaDirectory` wordt al verwijderd in die `each`. Lijn 161 mag dus weg denk ik

File: src/MediaCollections/Filesystem.php [hunk 27]
```diff
@@ -133,66 +150,65 @@ public function copyFromMediaLibrary(Media $media, string $targetFile): string
...
     {
         $newFileName = $media->file_name;
         $oldFileName = $media->getOriginal('file_name');
 
         $mediaDirectory = $this->getMediaDirectory($media);
 
-        $oldFile = $mediaDirectory.'/'.$oldFileName;
-        $newFile = $mediaDirectory.'/'.$newFileName;
+        $oldFile = $mediaDirectory . '/' . $oldFileName;
+        $newFile = $mediaDirectory . '/' . $newFileName;
```
Comment (AlexVanderbist) [inline src/MediaCollections/Filesystem.php]:
```php
$oldFile = "{$mediaDirectory}/{$oldFileName}";
$newFile = "{$mediaDirectory}/{$newFileName}";
```

File: src/MediaCollections/HtmlableMedia.php [hunk 28]
```diff
@@ -0,0 +1,99 @@
...
+    }
+
+    public function toHtml()
+    {
+        if (! (new Image())->canHandleMime($this->media->mime_type)) {
+            return '';
+        }
+
+        $attributeString = collect($this->extraAttributes)
+            ->map(fn($value, $name) => $name.'="'.$value.'"')->implode(' ');
```
Comment (AlexVanderbist) [inline src/MediaCollections/HtmlableMedia.php]: Do we need to encode HTML entities here?

Comment (AlexVanderbist) [inline src/MediaCollections/HtmlableMedia.php]: Do we need to encode HTML entities here?

File: docs/introduction.md [hunk 29]
```diff
@@ -3,34 +3,36 @@ title: Introduction
 weight: 1
 ---
 
-Medialibrary is a Laravel (5.6 and up) package that can associate all sorts of files with Eloquent models. It provides a simple, fluent API to work with.
+**V8 is not production ready, a stable version will be released soon**
+
+MediaLibrary is a Laravel (5.6 and up) package that can associate all sorts of files with Eloquent models. It provides a simple, fluent API to work with.
```
[the remark this answers is not in the document]
Comment (freekmurze) [reply docs/introduction.md]: Fixed.

File: src/Conversions/Commands/RegenerateCommand.php [hunk 30]
```diff
@@ -1,47 +1,40 @@
...
+use Spatie\MediaLibrary\Conversions\FileManipulator;
+use Spatie\MediaLibrary\MediaCollections\MediaRepository;
+use Spatie\MediaLibrary\MediaCollections\Models\Media;
 
 class RegenerateCommand extends Command
 {
     use ConfirmableTrait;
 
-    protected $signature = 'medialibrary:regenerate {modelType?} {--ids=*}
+    protected $signature = 'media-library:regenerate {modelType?} {--ids=*}
```
[the remark this answers is not in the document]
Comment (freekmurze) [reply src/Conversions/Commands/RegenerateCommand.php]: Added.

File: src/Conversions/Conversion.php [hunk 31]
```diff
@@ -265,13 +211,22 @@ public function getResultExtension(string $originalFileExtension = ''): string
         return $originalFileExtension;
     }
 
-    public function getConversionFile(string $file): string
+    public function getConversionFile(Media $media): string
+    {
+        return
+            $this->conversionFileNamer->getFileName($this, $media)
+            . '.' . $this->conversionFileNamer->getExtension($this, $media);
```
[the remark this answers is not in the document]
Comment (freekmurze) [reply src/Conversions/Conversion.php]: Great suggestion, thanks!

File: src/Conversions/ConversionCollection.php [hunk 32]
```diff
@@ -145,27 +111,22 @@ protected function addManipulationToConversion(Manipulations $manipulations, str
         }
 
         if ($conversionName === '*') {
-            $this->each->addAsFirstManipulations(clone $manipulations);
+            $this
+                ->each(fn(Conversion $conversion) => $conversion->addAsFirstManipulations(clone $manipulations));
```
[the remark this answers is not in the document]
Comment (freekmurze) [reply src/Conversions/ConversionCollection.php]: Done!

File: src/Conversions/ConversionCollection.php [hunk 33]
```diff
@@ -93,15 +65,12 @@ protected function addConversionsFromRelatedModel(Media $media)
...
-    /**
-     * Add the extra manipulations that are defined on the given media.
-     *
-     * @param \Spatie\MediaLibrary\Models\Media $media
-     */
     protected function addManipulationsFromDb(Media $media)
     {
         collect($media->manipulations)->each(function ($manipulations, $conversionName) {
-            $this->addManipulationToConversion(new Manipulations([$manipulations]), $conversionName);
+            $manipulations = new Manipulations([$manipulations]);
```
[the remark this answers is not in the document]
Comment (freekmurze) [reply src/Conversions/ConversionCollection.php]: 👍

File: src/Conversions/FileManipulator.php [hunk 34]
```diff
@@ -1,41 +1,31 @@
...
+    public function createDerivedFiles(Media $media, array $only = [], bool $onlyMissing = false): void
     {
         $profileCollection = ConversionCollection::createForMedia($media);
 
-        if (! empty($only)) {
-            $profileCollection = $profileCollection->filter(function ($collection) use ($only) {
-                return in_array($collection->getName(), $only);
-            });
+        if (!empty($only)) {
+            $profileCollection = $profileCollection->filter(fn($collection) => in_array($collection->getName(), $only));
```
[the remark this answers is not in the document]
Comment (freekmurze) [reply src/Conversions/FileManipulator.php]: Good catch, fixed!

File: src/Conversions/FileManipulator.php [hunk 35]
```diff
@@ -51,37 +41,30 @@ public function createDerivedFiles(Media $media, array $only = [], bool $onlyMis
...
-            $temporaryDirectory->path(Str::random(16).'.'.$media->extension)
+            $temporaryDirectory->path(Str::random(16) . '.' . $media->extension)
         );
 
         $conversions
             ->reject(function (Conversion $conversion) use ($onlyMissing, $media) {
                 $relativePath = $media->getPath($conversion->getName());
 
-                $rootPath = config('filesystems.disks.'.$media->disk.'.root');
+                $rootPath = config('filesystems.disks.' . $media->disk . '.root');
```
[the remark this answers is not in the document]
Comment (freekmurze) [reply src/Conversions/FileManipulator.php]: 👍

File: src/Conversions/FileManipulator.php [hunk 36]
```diff
@@ -124,10 +110,10 @@ public function performManipulations(Media $media, Conversion $conversion, strin
...
         }
 
-        $conversionTempFile = pathinfo($imageFile, PATHINFO_DIRNAME).'/'.Str::random(16)
-            .$conversion->getName()
-            .'.'
-            .$media->extension;
+        $conversionTempFile = pathinfo($imageFile, PATHINFO_DIRNAME) . '/' . Str::random(16)
+            . $conversion->getName()
+            . '.'
+            . $media->extension;
```
[the remark this answers is not in the document]
Comment (freekmurze) [reply src/Conversions/FileManipulator.php]: done!

File: src/Conversions/Jobs/PerformConversionsJob.php (hunk 16 above)
Comment (freekmurze) [reply src/Conversions/Jobs/PerformConversionsJob.php]: yes!

File: src/Conversions/Jobs/PerformConversionsJob.php (hunk 17 above)
Comment (freekmurze) [reply src/Conversions/Jobs/PerformConversionsJob.php]: 👍

File: src/MediaCollections/Commands/ClearCommand.php (hunk 19 above)
Comment (freekmurze) [reply src/MediaCollections/Commands/ClearCommand.php]: Done!

**Comment by freekmurze** [comment]
Great code review, thanks @AlexVanderbist 👍
